### ISSUE TITLE
tf.keras.layers.Input with ragged=True is not setting uniform outer dimensions

### ISSUE THREAD

**Issue by alexbaras** [tensorflow/tensorflow#47771]
**System information**
- Have I written custom code: Yes, but this is expected functionality of tf.keras.layers.Input for ragged input
- OS Platform and Distribution: Linux Ubuntu 20 LTS
- TensorFlow installed from: binary
- TensorFlow version: 2.2.0
- Python version: 3.8.5
- Bazel version: N/A
- GCC/Compiler version: N/A
- CUDA/cuDNN version: 11.0
- GPU model and memory: V100 / 32G

**Describe the current behavior**
```
input_tiles = tf.keras.layers.Input(shape=(2, None, 128, 128, 3), dtype=tf.float32, ragged=True)
input_tiles.shape
Out[97]: TensorShape([None, None, None, 128, 128, 3])
```

**Describe the expected behavior**
```
input_tiles = tf.keras.layers.Input(shape=(2, None, 128, 128, 3), dtype=tf.float32, ragged=True)
input_tiles.shape
Out[97]: TensorShape([None, 2, None, 128, 128, 3])
```

More specifically, there should be only 1 ragged axis since there was only 1 dim listed as None in the shape param of tf.keras.layers.Input with ragged=True.

[Tensorflow Documentation](https://www.tensorflow.org/api_docs/python/tf/keras/Input)
"A boolean specifying whether the placeholder to be created is ragged. Only one of 'ragged' and 'sparse' can be True. In this case, values of 'None' in the 'shape' argument represent ragged dimensions. For more information about RaggedTensors, see this guide."

**Standalone code to reproduce the issue**
See above.

**Comment by Saduf2019** [comment]
@alexbaras 
This has been fixed in tf 2.4, please refer to the [gist here](https://colab.research.google.com/gist/Saduf2019/9950829dc1dd83a16d3e7161f78fb146/untitled561.ipynb)

**Comment by alexbaras** [comment]
Ok thanks
